api/coreservice.go

```diff
@@ -1514,11 +1514,22 @@ func (core *coreService) EstimateMigrateStakeGasConsumption(ctx context.Context,
 		GasLimit:       g.BlockGasLimitByHeight(header.Height() + 1),
 		Producer:       zeroAddr,
 	})
+
 	exec, err := staking.FindProtocol(core.registry).ConstructExecution(ctx, ms, core.sf)
 	if err != nil {
 		return 0, err
 	}
-	gas, err := core.EstimateExecutionGasConsumption(ctx, exec, caller)
+	gas, err := core.EstimateExecutionGasConsumption(ctx, exec, caller, protocol.WithSimulatePreOpt(func(sm protocol.StateManager) error {
```

Comment:
can we do this by adding a new `ConstructExecutionForSimulation`?
```
func ConstructExecutionForSimulation() {
	ConstructExecution()
	// add amount to the sender account
	sender, err := accountutil.LoadAccount(sm, caller)
	if err != nil {
		return err
	}
	if err = sender.AddBalance(exec.Amount()); err != nil {
		return err
	}
	return accountutil.StoreAccount(sm, caller, sender)
}
```
this way we can avoid adding `SimulationConfig`, which works but like an overkill

Reply:
already discussed offline